app/controllers/api/v1/conversations_controller.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling conversation requests.
+    class ConversationsController < ApplicationController
+      include DeviseTokenAuth::Concerns::SetUserByToken
+      include Pundit
+
+      before_action :authenticate_user!
+      before_action :find_conversation, only: %i[show destroy update]
+      before_action :find_conversation_id, only: %i[add_user delete_user]
+      before_action :authorize_conversation!
+      before_action :verify_authorized
+
+      def index
+        @conversations = current_user.conversations
+        render json: @conversations, status: :ok
+      end
+
+      def show
+        if @conversation
+          render json: @conversation, status: :ok
+        else
+          render json: @conversation.errors, status: :not_found
+        end
+      end
+
+      def add_user
+        user = User.find_by(id: params[:id])
+        if user
+          @conversation.users << user
```

Comment:
what is the response here?
why don't we validate if user was assigned?

Reply:
we also need to check if conversation exists and only after that try to assign user